### ISSUE TITLE
HTML step attribute

### ISSUE THREAD

**Issue by michael-n-cooper** [w3c/aria#471]
Need feature for step attribute. https://www.w3.org/TR/html51/sec-forms.html#the-step-attribute

Tracker issue: https://www.w3.org/WAI/ARIA/track/issues/539

**Comment by cookiecrook** [comment]
Add aria-valuestep for feature parity with step attr on HTML 5 range input (slider). There is currently no way to programmatically determine an acceptable value within a range, so otherwise sliders implementing valueChangeRequest would need to accept any floating point value between min and max.

**Comment by cookiecrook** [comment]
If ARIA 1.2 is still limited to HTML Role Parity, this issue is not in scope.

**Comment by jnurthen** [comment]
also https://www.w3.org/Bugs/Public/show_bug.cgi?id=27870

**Comment by jnurthen** [comment]
So it seems that UIA and ATK map this - 

UIA | If the input is in the Range state, set both RangeValue.SmallChange and RangeValue.LargeChange to the value of step.
Exposed as atk_value_get_minimum_increment if the element implements the AtkValue interface.

so for completeness we should probably do this.

Anyone want to do this in 1.3? If so please assign yourself.

**Comment by scottaohara** [comment]
Will throw my hat in for a try at this.

**Comment by joanmarie** [comment]
While ATK has this, Orca doesn't use it. What is the use case of step and who would consume it?

**Comment by joanmarie** [comment]
i.e. Does nsAccessibility support this and VoiceOver need it?

**Comment by jnurthen** [comment]
@cookiecrook should we close?

**Comment by cookiecrook** [comment]
Yes, but for posterity, here's a somewhat-related scenario.

- Some site (don't recall which) used a native slider with 3–4 values, and used JavaScript to enforce the slider thumb would land on the closest "step", but they didn't use the native `step` attr. Yes, I agree "Don't Do That™", but here's the scenario.
- Let's call it 4 total steps for illustration. Each value change equals 25%.
- VoiceOver-triggered steps default to about 10%, which is less than half of the 25% custom "step" delta.
- The Slider started at 0%, next closest "step" at 25%.
- VoiceOver user-triggered increment sets the slider value to 10%.
- JavaScript detects the value change, then enforces locking it to the nearest value, 0% (not the 25% user intention).
- So it seems like the slider is broken. "I've incremented from 0, but it's still set to 0."

So if custom ARIA sliders are ever able to allow more granular control (`setValue` or similar), `aria-valuestep` may be worth revisiting. Currently it does not seem like that will happen anytime soon (maybe never), since we can get by with simulated keypresses (arrows, pageup, etc.)

TLDR: I am okay closing this issue, with the understanding that it might be necessary later if something like a setValue API is implemented.
